docs/source/guides/integrations.mdx

```diff
@@ -0,0 +1,279 @@
+3. **Inference API:** use our servers to run inference on models supported by your library for free.
+4. **Widgets:** display a widget on the landing page of your models on the Hub. It allows users to quickly try a model
+   from the browser.
+
+In this guide, we will focus on the first two topics. If you are interested in Inference and Widgets, you can follow
+[this guide](https://huggingface.co/docs/hub/models-adding-libraries#set-up-the-inference-api). In both cases, you can
+reach out to us if you are integrating your library with the Hub and want to be listed [in our docs](https://huggingface.co/docs/hub/models-libraries). 
+
+## A first approach: helpers
+
+The first approach to integrate a library to the Hub is to actually implement the `push_to_hub` and `from_pretrained`
+methods by yourself. This gives you full flexibility on which files you need to upload/download and how to handle inputs
+specific to your framework. You can refers to the two [upload files](./upload) and [download files](./download) guides
+to learn more on how to do that. This is for example how FastAI integration is implemented (see [`push_to_hub_fastai`]
+and [`from_pretrained_fastai`]).
+
+Implementation can defer between libraries but the workflow is often similar.
+
+### from_pretrained
+
+This is how a `from_pretrained` method usually look like:
+
+```python
+def from_pretrained(model_id: str):
+   # Download model from Hub
+   cached_model = hf_hub_download(
+      repo_id=repo_id,
+      filename="model.pkl",
+      library_name="fastai",
+      library_version=get_fastai_version(),
+   )
+
+   # Load model
+    return load_model(cached_model)
+```
+
+### push_to_hub
+
+The `push_to_hub` often require a bit more complexity to handle repo creation, generate the model card and save weights.
+A common approach is to save all of these files in a temporary folder, upload it and then delete it.
+
+```python
+def push_to_hub(model, repo_name: str) -> None:
+   api = HfApi()
+
+   # Create repo if not existing yet and get the associated repo_id
+   repo_id = api.create_repo(repo_name, exist_ok=True)
+
+   # Save all files in a temporary directory and push them in a single commit
+   with TemporaryDirectory() as tmpdir:
+      tmpdir = Path(tmpdir)
+
+      # Save weights
+      save_model(model, tmpdir / "model.safetensors")
+
+      # Generate model card
+      card = generate_model_card(model)
+      (tmpdir / "README.md").write_text(card)
+
+      # Save logs
+      # Save figures
+      # Save evaluation metrics
+      # ...
+
+      # Push to hub
+      return api.upload_folder(repo_id=repo_id, folder_path=tmpdir)
+```
+
+This is of course only an example. If you are interested in more complex manipulations (delete remote files, upload
+weights on the fly, persist weights locally,...) please refer to the [upload files](./upload) guide.
+
+### Limitations
```

Comment:
I'd move the Limitations section above the `from_pretrained` method. I think it'd be nice to have a full picture of the pros/cons of using this approach before actually trying to implement it.

Reply:
I understand your point but I think it could be confusing to users -at least with the text I wrote-. If I move the "limitations" section up, the reader would not really have in mind the method signatures/implementation that I already list everything that is missing from it.

What I can do is to add a sentence in the introduction to say "hey, we will see 2 different approaches, each of them have advantages/drawbacks so choose what's best for you". So that when the user reads the first approach he/she already have in mind that there are some limitations (without knowing which ones yet). What do you think?

Reply:
Great idea! 👍